hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -609,11 +603,8 @@ private void writeChunkToContainer(ChunkBuffer chunk) throws IOException {
         }
         return e;
       }, responseExecutor).exceptionally(e -> {
-        if (LOG.isDebugEnabled()) {
-          LOG.debug(
-              "writing chunk failed " + chunkInfo.getChunkName() + " blockID "
-                  + blockID + " with exception " + e.getLocalizedMessage());
-        }
+        LOG.debug("writing chunk failed {} blockID {} with exception {}",
```

Comment:
This is part of the I/O path where we have seen perf issue without the if (LOG.isDebugEnabled()), which slf4j claim that parameterize log will solve. So my suggestion is to keep the  if (LOG.isDebugEnabled()) and ignore the Sonar warning if any here.

Reply:
Yes, we need to keep the explicit check for log level. The lazy evaluation only applies to the toString() call, the expression will be fully evaluated if any provided as a param, e.g.: e.getLocalizedMessage() above.